### ISSUE TITLE
Custom Docker image for the template

### ISSUE THREAD

**Issue by oleg-nenashev** [hitsumabushi845/MkDocs-with-Remote-Containers#10]
Hi @hitsumabushi845. As you have already noticed, I used this template to build a DevContainer for the upcoming Gradle Community site: https://github.com/gradle/community/pull/8 . This template works really well, and it would be awesome to convert it to the official DevContainer listed within VSCode & Co!

While applying the container, I noticed a few issues related to the base image:

- There is no `libsass` and the `gcc` toolchain  that is needed for the `mkdocs-extra-sass-plugin`
- There is no `git` package that is needed for the `mkdocs-multirepo-plugin`
- There is no `libcairo2 ` required for the `social` plugin which is a part of Material for MkDocs

In my pull request I used a small custom image to address those problems. I wonder whether there would be an interest to integrate those patches to upstream and, maybe, to work with the Microsoft team to have it as a template DevContainer definition

**Comment by hitsumabushi845** [comment]
@oleg-nenashev 
Thanks for reporting an issue for my repo:) I answer your reporting issues.

> - There is no libsass and the gcc toolchain that is needed for the mkdocs-extra-sass-plugin
> - There is no git package that is needed for the mkdocs-multirepo-plugin
> - There is no libcairo2  required for the social plugin which is a part of Material for MkDocs

This base image is created to use only mkdocs.  
The reason is that to consider base image can treat all of 3rd-party plugins, the size will increase. 
So, users must modify base image to use material-mkdocs or any other plugins currently.  

But, I'm interested to improve this template repository and make this repo to official devcontainer template for VSCode if I can.

**Comment by oleg-nenashev** [comment]
Hi, thanks for the response. I agree about the complexity of external plugins. And, indeed, being able to change the image, like I did, is a feasible solution

The image size does not seem to be a huge concern, though. This image is for local development, so its performance implications are secondary to developer experience as long as it runs well on the stock developer machines.

**Comment by hitsumabushi845** [comment]
> The image size does not seem to be a huge concern, though. This image is for local development, so its performance implications are secondary to developer experience as long as it runs well on the stock developer machines.

I agree with that opinion. However, we need to consider the extent to which 3rd party plugins should be supported in a template Docker image.

I think we should support at least the `git` package to use git in dev containers. Other packages need to be considered separately.

**Comment by oleg-nenashev** [comment]
FWIW, to add an official template, a PR would be needed in https://github.com/devcontainers/templates

**Comment by hitsumabushi845** [comment]
Thanks, I will start by creating a better Dockerfile in this repository to contribute to the official template.

**Comment by hitsumabushi845** [comment]
I've created a PR https://github.com/hitsumabushi845/MkDocs-with-Remote-Containers/pull/11 to work with this issue.

Feel free to comment on this PR if you have any requests.
